test/sql/testing_sql_util.cpp

```diff
@@ -69,6 +93,12 @@ ResultType TestingSQLUtil::ExecuteSQLQueryWithOptimizer(
     std::string &error_message) {
   auto &peloton_parser = parser::PostgresParser::GetInstance();
   std::vector<type::Value> params;
+  // # 623
+  auto& txn_manager = concurrency::TransactionManagerFactory::GetInstance();
+  auto txn = txn_manager.BeginTransaction();
+  traffic_cop_.tcop_txn_state_.emplace(txn, ResultType::SUCCESS);
+  traffic_cop_.single_statement_txn = true;
+  optimizer->consistentTxn = txn;
```

Comment:
I think it is thread-safe. one worker thread has several connections, one connection has one tcop and one tcop has one transaction.